addons/account/models/account_journal_dashboard.py

```diff
@@ -349,12 +349,35 @@ def _fill_bank_cash_dashboard_data(self, dashboard_data):
                    ) statement ON TRUE
              WHERE journal.id = ANY(%s)
                AND statement.company_id = ANY(%s)
-        """, [self.ids, self.env.companies.ids])
+        """, [bank_cash_journals.ids, self.env.companies.ids])
         last_statements = {journal_id: statement_id for journal_id, statement_id in self.env.cr.fetchall()}
-        self.env['account.bank.statement'].browse(i for i in last_statements.values() if i).mapped('balance_end_real')  # prefetch
+        self.env['account.bank.statement'].browse(i for i in last_statements.values() if i).mapped(lambda s: (s.balance_end_real, s.date))  # prefetch
```

Comment:
Isn't the date prefetched with the rest anyways?

Reply:
Note to self: all stored fields are prefetched (except id)